INSTALLATION.MD

```diff
@@ -0,0 +1,93 @@
+# Installation
+This is tested under Ubuntu 20.04 LTS
+
+We're assuming, that the ticker api will be available under the `api.domain.tld` domain. Please change accordingly.
+
+## Requirements:
+- `nginx`
+- certificate (`certbot` & `python3-certbot-nginx` to use free Let's Encrypt Certs)
+- `git`
+- Public IPv4
+- Public IPv6 (Please!)
+- `make`
+
+
+### Getting Go:
+_Don't use the shipped version of your system, if you're working on a Debian based OS (Ubuntu, etc)_ 
+
+Instead use:
+ [golang.org install guide](https://golang.org/doc/install)
+
+## Installation
+
+1. `cd /var/www/`
+The directory where we gonna install all the things
+1. `git clone --recurse-submodules https://github.com/systemli/ticker`
+Clone the repository onto your disk
+1. `cd ticker`
+Go into the just cloned repository
+3. `git submodule update --init --recursive`
+Initialise all the submodules
+1. `vim config.yml.dist`
+Fill your config file with the following content:
+```yaml
+# listen binds ticker to specific address and port
+listen: "localhost:8080"
+# log_level sets log level for logrus
+log_level: "error"
+# initiator is the email for the first admin user (see password in logs)
+initiator: "<your e-mail>"
+# database is the path to the bolt file
+database: "ticker.db"
+# secret used for JSON Web Tokens
+secret: "<your special little secret> (make it LOOOONG!)"
+# twitter configuration
+twitter_consumer_key: ""
+twitter_consumer_secret: ""
+# listen port for prometheus metrics exporter
+metrics_listen: ":8181"
+# path where to store the uploaded files
+upload_path: "uploads"
+# base url for uploaded assets
+upload_url: "https://api.domain.tld"
+```
+6. `make run`
+Run the application
+
+__Done__
+
+## Exposing
+`vim /etc/nginx/sites-available/ticker-api`
+
+
+```nginx.conf
```

Comment:
I would redirect all http traffic to https with hsts headers. You mentioned later how to obtain a valid certificate with certbot and encrypted connections should be the default.

Reply:
The config example throughout the whole docs are written to listen to port tcp/80 and using certbot with the nginx flag to automatically reconfigure the webserver to listen to tcp/443 & deploying best practice security standards.

IMHO this has the benefit of not needing to implement the config & change the tutorial everytime a best practice changes, because it's automatically done by certbot.

I believe that people not using this guideline have the needed knowledge to configure an nginx accordingly.

Reply:
In https://github.com/systemli/ticker/pull/44/commits/e6e896e42f7a1bfede9b0c1a1ea87ca66a1592df I added the note to implement TLS best practice accordingly if not using certbot.

Additionally in https://github.com/systemli/ticker/pull/44/commits/2473d5dd7b548c1fd48180119c1b37e0e3f3e7e7 I added the `--redirect` flag to the certbot commands, so there will be an automatic redirect from HTTP connections to an secure HTTPS connection via a 301 (looking like this:

```nginx.conf
server {

	server_name api.domain.tld;

	location / {
		proxy_pass http://127.0.0.1:8080;
		proxy_set_header Host $host;
		proxy_set_header X-Real-IP $remote_addr;

	}

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/api.domain.tld/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/api.domain.tld/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = api.domain.tld) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	listen 80 default_server;
	listen [::]:80 default_server;

	server_name api.domain.tld;
    return 404; # managed by Certbot


}
```